resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterManager.scala

```diff
@@ -81,6 +85,27 @@ private[spark] class KubernetesClusterManager extends ExternalClusterManager wit
         configMapKey)
     }
 
+    val hadoopBootStrap = for {
+      hadoopConfigMap <- maybeHadoopConfigMap
+    } yield {
+      val hadoopUtil = new HadoopUGIUtil
+      val hadoopConfigurations = maybeHadoopConfDir.map(
+          conf_dir => getHadoopConfFiles(conf_dir)).getOrElse(Array.empty[File])
+      new HadoopConfBootstrapImpl(
+        hadoopConfigMap,
+        hadoopConfigurations,
+        hadoopUtil
+      )
+    }
+    val kerberosBootstrap = for {
```

Comment:
The Kerberos bootstrap should only happen if the hadoop bootstrap is provided, right? We probably want to have an error case that requires the Hadoop config map to be provided if Kerberos is requested.

Reply:
For the executor? That wouldn't happen as we only require the token to be present and the sparkConf to include the necessary kerberos config